### DISCUSSION TITLE
[any package] How to request a package to be held by choco-community?

### DISCUSSION THREAD

**Discussion by jo-chemla** [chocolatey-community/chocolatey-packages#2093] [Q&A]
Hello chocolatey-community, 
Once every few week I stumble upon a windows software which does not yet have a chocolatey package, but do find a stable way to install it - either via a standalone executable, zipped folder, or exe/msi installer present on the original soft website, github repo releases section, etc. 

I'd like to know if filling an issue on this [chocolatey-packages](https://github.com/chocolatey-community/chocolatey-packages) repo is the correct way to have the community build a package for it - even if I am just a user and not a developper of the aforementioned software. On the issue submission template, I see I can `Open a discussion about moving a package you own, or have permission to move over to this repository.`, but this is slightly different since I'm not the package owner. 

I do understand having the package built by choco-community is slightly different than having the original project maintainer or team builds the choco package for example using Github Actions or the AU, which would require less manual steps. I've crawled through the docs to see if I could fill an issue or submit a PR but am not 100% confident on the way to do so. 
Thanks a lot for your feedback!

**Comment by TheCakeIsNaOH** [comment]
The packages in this repository (`chocolatey-community/chocolatey-packages`) are packages that are maintained by a core group of maintainers (and by anyone else who wants to contribute). On the Chocolatey Community Repository, if you packages maintained by the `chocolatey-community` maintainer, they are coming from this GitHub repository (or one of a couple others under this organization). 

If you want to have a new package added here, then opening a discussion is the correct thing to do.

But if you want to have a new package added to the Chocolatey Community Repository (`community.chocolatey.org`) in general by anyone who volunteers, then open an issue at the package requests repository: https://github.com/chocolatey-community/chocolatey-package-requests

**Reply by jo-chemla** [reply]
Thanks for the clarification. So there are two ways to request a new package to be published by the chocolatey-community:
 -  open a discussion on `chocolatey-community/chocolatey-packages` which would be picked up by a core group of maintainers (+ open to contributions)
 - fill an RFP issue on  `chocolatey-community/chocolatey-package-requests`, to be picked-up by anyone

Is one of them preferred over the other? Thanks again!

**Reply by pauby** [reply]
There is no preference.

I'm trying to think of what criteria you'd use to pick one over the other. @AdmiringWorm may be a better person to answer this, but I'm going to give it a go.

This repository is for the Chocolatey Community Chocolatey Packages. It relies on the community to maintain those packages. If you were interested in adding the package to the repository yourself, and get some help to add it through the PR you put in, then I'd suggest adding a discussion here would be good.

If you have no interest at all in creating a package, then the package requests repository would be the best place.

**Reply by AdmiringWorm** [reply]
The preferred way is to open an issue in the chocolatey-package-requests repository. If the maintainer that then picks up the issue wants to move to this repository, they should then open up a discussion to migrate the package here. In general, we do not accept entirely new packages in this repository (with a few exceptions), and only accept migrating packages.

**Comment by photodude** [comment]
@AdmiringWorm do we need a maintainer to get a package included?
I submitted an RFP request a long time ago https://github.com/chocolatey-community/chocolatey-package-requests/issues/118 for PHPmyadmin but it hasn't gotten picked up. 

but in the Age of AI I was thinking I might be able to create the initial submission files for getting this package included. 

``` XML
<?xml version="1.0"?>
<package>
  <metadata>
    <id>phpmyadmin</id>
    <version>5.1.1</version>
    <title>PHPMyAdmin</title>
    <authors>PHPMyAdmin Team</authors>
    <owners>PHPMyAdmin Team</owners>
    <summary>PHPMyAdmin is a free software tool written in PHP intended to handle the administration of MySQL over the World Wide Web.</summary>
    <description>PHPMyAdmin is a free software tool written in PHP intended to handle the administration of MySQL over the World Wide Web.</description>
    <projectUrl>https://www.phpmyadmin.net/</projectUrl>
    <licenseUrl>https://www.phpmyadmin.net/license/</licenseUrl>
    <tags>mysql php database administration</tags>
  </metadata>
  <files>
    <file src="https://files.phpmyadmin.net/phpMyAdmin/5.1.1/phpMyAdmin-5.1.1-all-languages.zip" target="tools" />
  </files>
</package>
```

```bash
$toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
$zipPath = Join-Path $toolsDir "phpMyAdmin.zip"
(New-Object System.Net.WebClient).DownloadFile('https://files.phpmyadmin.net/phpMyAdmin/5.1.1/phpMyAdmin-5.1.1-all-languages.zip', $zipPath)
Expand-Archive $zipPath $toolsDir
Remove-Item $zipPath
Write-Host "PHPMyAdmin installed to $($toolsDir)phpMyAdmin-5.1.1-all-languages"
```

But I'm also not sure if that's all that's needed. The submission guidelines and instructions are not easy for me to follow.

**Reply by AdmiringWorm** [reply]
Yes, a maintainer is required.

Your AI-generated script would be a Hard no here in this repository, and it would not work as a Chocolatey CLI package.

If you'd like to try creating a package, you should start with getting the basics done before wrapping your head around the guidelines here.

Most package creation documentation is part of the package generated when you run `choco new packageName`, so I recommend starting there.
